Upcoming:[DI-23319] - CloudPulse Alerting minor UX improvements

venkymano-akamai opened the pull request:
## Description 📝

Minor UX improvements of CloudPulse Alerting across, create alert flow , edit default alert flow and resources section

## Changes  🔄

1. Fixed width for the Add dimension Filter button in the create alert flow.
2. Wait for the API to complete the response in the edit default alert resources flow.
3. Zeroth state error message update in the resources section of alerting.
4. Reduce spacing between the different sections of the cloudpulse alerting resources section.
5. Show snackbar at the bottom of the screen instead of top

## Target release date 🗓️

11-03-2025

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2025-02-27 at 4 04 57 PM](https://github.com/user-attachments/assets/02334bcf-1591-4c12-a474-3ba80ddfc0fb)|![Screenshot 2025-02-27 at 3 41 22 PM](https://github.com/user-attachments/assets/f184bd75-6d19-436e-865e-2aa6c52efad8)|
|![Screenshot 2025-02-27 at 4 05 49 PM](https://github.com/user-attachments/assets/affefaa6-b228-4489-8978-5b5b90754b35)|![Screenshot 2025-02-27 at 3 43 04 PM](https://github.com/user-attachments/assets/f448a4f5-7459-4b43-987b-ad7af7c5655b)|
|![Screenshot 2025-02-27 at 4 09 10 PM](https://github.com/user-attachments/assets/b3894d48-765c-47cd-856e-8959e4ef0d54)|![Screenshot 2025-02-27 at 3 40 17 PM](https://github.com/user-attachments/assets/ddfd6205-e214-4c16-bc7a-2e14ba39697e)|
|![Screenshot 2025-02-27 at 4 05 59 PM](https://github.com/user-attachments/assets/e1f19721-7dc1-4940-a964-f54a0ecfb2d9)|![Screenshot 2025-02-27 at 4 11 41 PM](https://github.com/user-attachments/assets/9a684a54-ed98-4267-8829-6e01f286f0e2)|

## How to test 🧪

1. Login as legacy mock since some endpoints are not available.
2. Navigate to Monitor tab and then to alerts section.

To validate **Fixed width for the Add dimension Filter button**

1. Open the create alert flow, you will be able to find a Add dimension filter button with 150px width

To validate **Wait for the API to complete the response in the edit default alert resources flow**

1. In the serverHandler.ts file, for this matcher **http.get(
    '*/monitor/services/:serviceType/alert-definitions/:id'** change the type to 'system' from 'user'
2. Now from the list alert page, choose an alert action and click on edit.
3. Add or remove the resources in the edit page and click submit, you might see a loading state in the confirmation dialog until the API call is finished.
4. After saving when redirecting to list page, you can see the snackbar at the bottom of the page

To Validate **Zeroth state error message update in the resources section of alerting.**

1. In the serverHandler.ts file, for this matcher **http.get(
    '*/monitor/services/:serviceType/alert-definitions/:id'** in the response add property **entity_ids:[]**
2. From list alerts page, click on action menu and go to show details page to see the empty state.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

coliu-akamai reviewed (commented):
thank you! few things I noticed:

Unrelated - On the Alerts landing page, I noticed this for screen widths from 1280px-1360px:
![image](https://github.com/user-attachments/assets/e8917559-5883-49e1-9435-9740c1db8a2f)

[DI-23319] notes that for trigger conditions, the text shifts when there's an error. I'm still seeing this:
![image](https://github.com/user-attachments/assets/4282bab3-4735-4246-8832-d07ccfff8e1c)
![image](https://github.com/user-attachments/assets/8f5b0d50-741d-4d2e-9991-47d588547f0b)

venkymano-akamai commented:
@coliu-akamai , fixed both, 
1. Added a flexWrap for the alertlist view, so that on no space, it will automatically wrap.
2. For trigger conditions, added a fixed height by surrounding it in a box

This should fix both and should be good for a pass.

coliu-akamai reviewed (approved):
Confirming UX issues fixed - thanks!

One small thing with the trigger error messaging:
![image](https://github.com/user-attachments/assets/5d0b59cd-bb20-4966-8836-cdb76e2bb0e7)

Maybe it should say something like "The value must be greater than 0."

venkymano-akamai commented:
Merging this since we have enough approvals, UT's pass, jobs and checks pass. The timerange-verification.spec.ts is not related to this PR, and it is tracked separately as well.
